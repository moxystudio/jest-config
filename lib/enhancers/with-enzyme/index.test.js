'use strict';

const withEnzyme = require('.');

const buildConfig = () => ({
    testEnvironment: 'jsdom',
    globals: {
        foo: 'bar',
    },
    setupFilesAfterEnv: ['foo'],
});

it('should match snapshot when target is web', () => {
    expect(withEnzyme('enzyme-adapter-react-16', 'web')(buildConfig())).toMatchSnapshot();
});

it('should match snapshot when target is react-native', () => {
    expect(withEnzyme('enzyme-adapter-react-16', 'react-native')(buildConfig())).toMatchSnapshot();
});

it('should throw if no adapter is passed', () => {
    expect(() => withEnzyme()).toThrow(/please provide an enzyme adapter/i);
});

it('should throw if target is invalid', () => {
    expect(() => withEnzyme('enzyme-adapter-react-16')).toThrow(/please provide a valid target/i);
});

it('should throw if testEnvironment is not jsdom', () => {
    const config = {
        ...buildConfig(),
        testEnvironment: 'node',
    };

    expect(() => withEnzyme('enzyme-adapter-react-16', 'web')(config)).toThrow(/expected testEnvironment to be set to/i);
});
