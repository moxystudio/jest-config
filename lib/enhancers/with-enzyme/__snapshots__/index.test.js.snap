// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when target is react-native 1`] = `
Object {
  "globals": Object {
    "__MOXY_JEST_CONFIG_ENZYME_ADAPTER__": "enzyme-adapter-react-16",
    "foo": "bar",
  },
  "setupFilesAfterEnv": Array [
    "foo",
    "<PROJECT_ROOT>/lib/enhancers/with-enzyme/setup.js",
    "<PROJECT_ROOT>/lib/enhancers/with-enzyme/setup-react-native.js",
  ],
  "testEnvironment": "jsdom",
}
`;

exports[`should match snapshot when target is web 1`] = `
Object {
  "globals": Object {
    "__MOXY_JEST_CONFIG_ENZYME_ADAPTER__": "enzyme-adapter-react-16",
    "foo": "bar",
  },
  "setupFilesAfterEnv": Array [
    "foo",
    "<PROJECT_ROOT>/lib/enhancers/with-enzyme/setup.js",
    "<PROJECT_ROOT>/lib/enhancers/with-enzyme/setup-web.js",
  ],
  "testEnvironment": "jsdom",
}
`;
